tests/python/relax/test_op_gradient_eliminate_collapse.py

```diff
@@ -0,0 +1,57 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from typing import Callable
+import tvm
+from tvm import relax
+from tvm.ir.op import Op
+from tvm.script import relax as R, tir as T
+import tvm.testing
+
+
+def _map_to_grad(op_func: Callable, op_name: str, grad_var, *args):
+    op = Op.get(op_name)
+    op_grad_func = op.get_attr("FPrimalGradient")
+    call = op_func(*args)
+    bb = relax.BlockBuilder()
+
+    with bb.function("main", list(args) + [grad_var]):
+        with bb.dataflow():
+            orig_var = bb.emit_output(call)
+            grad_call = relax.Tuple(op_grad_func(orig_var, call, grad_var, bb))
+        bb.emit_func_output(orig_var)
```

Comment:
Seems that the generated function is never used? What is the purpose here?

Reply:
Note that the gradient function need orig_var, and seems that creating such var and binding it to `call` must be done in a context of block builder

Reply:
I see. Seems creating a Function is too much. What about
* normalizing the Call by block builder
* creating a var
* updating the struct info of the var using `var._update_struct_info` https://github.com/mlc-ai/relax/blob/760dbe8a71361654fa64fa5f2d8a31f777f7d1ac/python/tvm/relax/expr.py#L684-L685

We don’t have to emit it, as here we only want a Var with the correct sinfo.